src/platform/ExtendedOTARequestorDriver.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <platform/ExtendedOTARequestorDriver.h>
+#include <platform/OTARequestorInterface.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+namespace {
+
+constexpr System::Clock::Seconds32 kUserConsentPollInterval = System::Clock::Seconds32(30);
+
+} // namespace
+
+void ExtendedOTARequestorDriver::UpdateAvailable(const UpdateDescription & update, System::Clock::Seconds32 delay)
+{
+    VerifyOrDie(mRequestor != nullptr);
+
+    if (update.userConsentNeeded == true && mUserConsentDelegate)
+    {
+        bool localConfigDisabled = false;
+        VerifyOrdo(DeviceLayer::ConfigurationMgr().GetLocalConfigDisabled(localConfigDisabled) == CHIP_NO_ERROR,
+                   ChipLogProgress(SoftwareUpdate, "Failed to get local config disabled, using as false"));
+
+        if (localConfigDisabled == false)
+        {
+            chip::ota::UserConsentSubject subject;
+            CHIP_ERROR err = GetUserConsentSubject(subject, update);
+            if (err != CHIP_NO_ERROR)
+            {
+                ChipLogError(SoftwareUpdate, "Failed to get user consent subject");
```

Comment:
I think all errors must be reported via the driver `HandleError` so that it can move forward with other actions.